From b01259471d71263fe6e840e436916e69073fcb07 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Mon, 22 Oct 2018 15:56:49 +0800
Subject: [PATCH 4901/5242] MLK-20008 ARM64: dts: i.MX8: disable flexspi0 for
 m4 usage

commit  afc1583d4be48b289562db497cc350fa060b381d from
https://source.codeaurora.org/external/imx/linux-imx.git

When M4 using XIP flexspi, need disable flexspi0 from device tree,
otherwise there will be conflict.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Acked-by: Ye Li <Ye.Li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts |    4 ++++
 .../boot/dts/freescale/fsl-imx8qxp-mek-m4.dts      |    4 ++++
 2 files changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts
index 1fbaca9..3fd9522 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-m4.dts
@@ -67,6 +67,10 @@
 	status = "disabled";
 };
 
+&flexspi0 {
+	status = "disabled";
+};
+
 &rpmsg1{
 	/*
 	 * 64K for one rpmsg instance, using 2 instance
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts
index 5372ef5..69aba7e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-m4.dts
@@ -102,3 +102,7 @@
 &flexcan2 {
 	status = "disabled";
 };
+
+&flexspi0 {
+	status = "disabled";
+};
-- 
1.7.9.5

